### PR TITLE
Add new Modal for Cognito Flow

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

Creates a new Modal to use Cognito Flows. Based on `ClaimRewardsButton`

### Dragons

N/A

### How Has This Been Tested?

Triggered manually by adding https://localhost:3002 to allowed URLs in the Cognito dashboard and running with HTTPS. Then, dispatching the event to the store with Redux Dev Tools.

### How will this change be monitored?

Keeps console logs/infos/errors, but adds prefix "COGNITO" for easier identification.

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-cognito-modal

File: src/pages/audio-rewards-page/components/modals/CognitoModal.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    flow.on('ui', (event: FlowUIOpenEvent | FlowUICloseEvent) => {
+      switch (event.action) {
+        case 'opened':
+          dispatch(setCognitoFlowStatus({ status: CognitoFlowStatus.OPENED }))
+          break
+        case 'closed':
+          dispatch(setCognitoFlowStatus({ status: CognitoFlowStatus.CLOSED }))
+          break
+        default:
```
Comment (dylanjeffers) [inline src/pages/audio-rewards-page/components/modals/CognitoModal.tsx:60]: wondering if we need a default here, i think if ts sees that 'opened' and 'closed' are the only types for `event.action`, then default is not needed?

File: src/pages/audio-rewards-page/components/modals/CognitoModal.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+          // nothing
+          break
+      }
+    })
+
+    flow.on('session', (event: FlowSessionEvent) => {
+      switch (event.action) {
```
Comment (dylanjeffers) [inline src/pages/audio-rewards-page/components/modals/CognitoModal.tsx:67]: same here!

File: src/pages/audio-rewards-page/components/modals/CognitoModal.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      }
+    })
+
+    flow.on('error', (event: FlowErrorEvent) => {
+      console.error(`COGNITO: Flow error: ${event.message}`)
+    })
+
+    flow.open()
+  }, [dispatch, handle])
+  useEffect(() => {
```
Comment (dylanjeffers) [inline src/pages/audio-rewards-page/components/modals/CognitoModal.tsx:96]: space :)

File: src/pages/audio-rewards-page/components/modals/CognitoModal.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    if (isOpen && scriptLoaded && handle) {
+      triggerFlow()
+    }
+  }, [triggerFlow, isOpen, scriptLoaded, handle])
```
Comment (dylanjeffers) [inline src/pages/audio-rewards-page/components/modals/CognitoModal.tsx:100]: i think we could benefit from using `useAsync` hook here to combine the useEffect and useCallback: https://github.com/streamich/react-use/blob/master/docs/useAsync.md

**Review by dylanjeffers — APPROVED** [review]
lglg

File: src/pages/audio-rewards-page/components/modals/CognitoModal.tsx (hunk 4 above)
Comment (rickyrombo) [reply src/pages/audio-rewards-page/components/modals/CognitoModal.tsx:100]: Ooh that's super cool!

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-cognito-modal

**Review by sddioulde — APPROVED** [review]
looks good

**Pull request merged**